examples/example_compiler_gym_service/service_cc/ExampleService.cc

```diff
@@ -29,179 +26,102 @@ template <typename T>
   return Status::OK;
 }
 
-std::vector<std::string> getBenchmarkUris() {
-  return {"benchmark://example-v0/foo", "benchmark://example-v0/bar"};
-}
+class ExampleCompilationSession final : public CompilationSession {
+ public:
+  using CompilationSession::CompilationSession;
 
-std::vector<ActionSpace> getActionSpaces() {
-  ActionSpace space;
-  space.set_name("default");
-  space.add_action("a");
-  space.add_action("b");
-  space.add_action("c");
+  std::string getCompilerVersion() const final override { return "1.0.0"; }
 
-  return {space};
-}
+  std::vector<ActionSpace> getActionSpaces() const final override {
+    ActionSpace space;
+    space.set_name("default");
```

Comment:
Shame these are protobufs, then they could just be params to the constructor.

Reply:
The problem is that this class is never instantiated by the user. They pass it into `CompilerGymService<T>` which handles lifetime management: 
https://github.com/facebookresearch/CompilerGym/blob/development/compiler_gym/service/runtime/CompilerGymServiceImpl.h#L64